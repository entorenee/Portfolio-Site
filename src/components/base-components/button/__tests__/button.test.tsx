import React from 'react'
import { render } from '@testing-library/react'

import Button from '..'

describe('<Button />', () => {
  it('renders a button correctly', () => {
    const { container } = render(<Button>Click me</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders an internal link correctly', () => {
    const { container } = render(<Button url='/about'>About</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders an anchor link correctly', () => {
    const { container } = render(
      <Button url='https://github.com/dlsemay'>My Github</Button>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
