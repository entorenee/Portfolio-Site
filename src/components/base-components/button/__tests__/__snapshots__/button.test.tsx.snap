// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a button correctly 1`] = `
.emotion-0 {
  background-color: #E6D8AF;
  color: #2D3B7F;
  padding: 0.2rem 1.3rem;
  border-radius: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-0:hover {
  color: #2D3B7F;
}

<button
  class="emotion-0"
  type="button"
>
  Click me
</button>
`;

exports[`<Button /> renders an anchor link correctly 1`] = `
.emotion-0 {
  background-color: #E6D8AF;
  color: #2D3B7F;
  padding: 0.2rem 1.3rem;
  border-radius: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-0:hover {
  color: #2D3B7F;
}

<a
  class="emotion-0"
  href="https://github.com/dlsemay"
  rel="noopener noreferrer"
  target="_blank"
>
  My Github
</a>
`;

exports[`<Button /> renders an internal link correctly 1`] = `
.emotion-0 {
  background-color: #E6D8AF;
  color: #2D3B7F;
  padding: 0.2rem 1.3rem;
  border-radius: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-0:hover {
  color: #2D3B7F;
}

<a
  class="emotion-0"
  href="/about"
>
  About
</a>
`;
